subprojects/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/DefaultGroovyJavaJointCompileSpec.java

```diff
@@ -43,4 +44,14 @@ public void setGroovyCompileOptions(GroovyCompileOptions compileOptions) {
     public void setGroovyClasspath(List<File> groovyClasspath) {
         this.groovyClasspath = groovyClasspath;
     }
+
+    @Override
+    public File getCompilationMappingFile() {
```

Comment:
Done.

Reply:
I added `Nullable` to getter and setter since [this test](https://github.com/gradle/gradle/blob/e9b14325173966aebc6c682ded7f5a5bff580784/subprojects/architecture-test/src/test/java/org/gradle/architecture/test/KotlinCompatibilityTest.java#L149) complains.

Reply:
The annotations should be added on the interface `GroovyCompileSpec` and consistently for getter and setter.